api4/post.go

```diff
@@ -169,7 +185,57 @@ func getPostsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		w.Header().Set(model.HEADER_ETAG_SERVER, etag)
 	}
 
-	w.Write([]byte(c.App.PreparePostListForClient(list).ToJson()))
+	c.App.AddCursorIdsForPostList(list, afterPost, beforePost, since, page, perPage)
+	clientPostList := c.App.PreparePostListForClient(list)
+
+	w.Write([]byte(clientPostList.ToJson()))
+}
+
+func getPostsForChannelAroundLastUnread(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireUserId().RequireChannelId()
+	if c.Err != nil {
+		return
+	}
+
+	userId := c.Params.UserId
+	if !c.App.SessionHasPermissionToUser(c.App.Session, userId) {
+		c.SetPermissionError(model.PERMISSION_EDIT_OTHER_USERS)
+		return
+	}
+
+	channelId := c.Params.ChannelId
+	if !c.App.SessionHasPermissionToChannel(c.App.Session, channelId, model.PERMISSION_READ_CHANNEL) {
+		c.SetPermissionError(model.PERMISSION_READ_CHANNEL)
+		return
+	}
+
+	fmt.Printf("\napi getPostsForChannelAroundLastUnread LimitBefore | LimitAfter: %+v %+v \n", c.Params.LimitBefore, c.Params.LimitAfter)
```

Comment:
is this needed?

Reply:
Sorry, not yet ready for dev review.

This is from recent commits where I added logging so I can inspect on PR build since it's not failing on my local machine.  Will ping you once ready.